pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/config/TableDataManagerConfig.java

```diff
@@ -85,94 +65,21 @@ public static TableDataManagerConfig getDefaultHelixTableDataManagerConfig(
-  }
-
-  public void overrideConfigs(String tableName, AbstractTableConfig tableConfig) {
-    IndexingConfig indexingConfig = tableConfig.getIndexingConfig();
-    _tableDataManagerConfig.setProperty(READ_MODE, indexingConfig.getLoadMode().toLowerCase());
-    _tableDataManagerConfig.setProperty(TABLE_DATA_MANAGER_NAME, tableConfig.getTableName());
 
-    List<String> invertedIndexColumns = indexingConfig.getInvertedIndexColumns();
-    if (invertedIndexColumns != null) {
-      _tableDataManagerConfig.setProperty(IndexLoadingConfigMetadata.KEY_OF_LOADING_INVERTED_INDEX,
-          invertedIndexColumns);
-    }
-    String starTreeFormat = indexingConfig.getStarTreeFormat();
-    if (starTreeFormat != null) {
-      _tableDataManagerConfig.setProperty(IndexLoadingConfigMetadata.KEY_OF_STAR_TREE_FORMAT_VERSION, starTreeFormat);
-    }
-    String columnMinMaxValueGeneratorMode = indexingConfig.getColumnMinMaxValueGeneratorMode();
-    if (columnMinMaxValueGeneratorMode != null) {
-      _tableDataManagerConfig.setProperty(IndexLoadingConfigMetadata.KEY_OF_COLUMN_MIN_MAX_VALUE_GENERATOR_MODE,
-          columnMinMaxValueGeneratorMode);
-    }
-
-    // Server configuration is always to DEFAULT or configured value
-    // Apply table configuration only if the server configuration is set with table config
-    // overriding server config
-    //
-    // Server config not set means table config has no impact. This provides additional
-    // security from inadvertent changes as both config will need to be enabled for the
-    // change to take effect
-    SegmentVersion tableConfigVersion =
-        SegmentVersion.fromString(indexingConfig.getSegmentFormatVersion(), SegmentVersion.DEFAULT_TABLE_VERSION);
-
-    SegmentVersion serverConfigVersion = SegmentVersion.fromString(
-        _tableDataManagerConfig.getString(IndexLoadingConfigMetadata.KEY_OF_SEGMENT_FORMAT_VERSION),
-        SegmentVersion.DEFAULT_SERVER_VERSION);
-
-    // override server based configuration with table level configuration iff table configuration
-    // is less than server configuration
-    if (SegmentVersion.compare(tableConfigVersion, serverConfigVersion) < 0) {
-      LOGGER.info("Overriding server segment format version: {} with table version: {} for table: {}",
-          serverConfigVersion, tableConfigVersion, tableName);
-      _tableDataManagerConfig.setProperty(IndexLoadingConfigMetadata.KEY_OF_SEGMENT_FORMAT_VERSION,
-          indexingConfig.getSegmentFormatVersion());
-    } else {
-      LOGGER.info("Loading table: {} with server configured segment format version: {}", tableName,
-          serverConfigVersion);
-    }
+    return new TableDataManagerConfig(defaultConfig);
   }
 
-  public IndexLoadingConfigMetadata getIndexLoadingConfigMetadata() {
-    return new IndexLoadingConfigMetadata(_tableDataManagerConfig);
+  public void overrideConfigs(@Nonnull AbstractTableConfig tableConfig) {
+    // Override table level configs
```

Comment:
I left this method here so that later if we need to override TableDataManagerConfig by TableConfig, here is the place to add code. Right now we don't need to override anything.